incubator/hnc/pkg/controllers/hierarchy_controller.go

```diff
@@ -82,6 +82,7 @@ type HierarchyReconciler struct {
 // namespace).
 type NamespaceSyncer interface {
 	SyncNamespace(context.Context, logr.Logger, string) error
+	SetupWithManager(manager ctrl.Manager) error
```

Comment:
Is it ok to put `SetupWithManager()` in the `NamespaceSyncer` interface too? It's needed because the return type of `createObjectReconciler()` is `NamesapceSyncer`.

Reply:
It can be removed after the new object controller replaces the old one.

Reply:
Technically, you should probably have a new interface called CommonObjectReconciler that _composes_ NamespaceSyncer and also adds SetupWithManager. Or you could duplicate the call to SetupWithManager twice in the function, that's not _too_ bad. 

See https://golang.org/pkg/io/#ReadWriter for how `ReadWriter` composes the `Reader` and `Writer` interfaces.

Reply:
Thanks for explaining and sharing the ReadWriter example! I added the `SetupWithManager()` calls into the new func for now, since I will change the new object controller `SetupWithManager()` to take an additional parameter of `maxReconciles` in the next commit.

Reply:
+1